dashboard/app/models/ability.rb

```diff
@@ -198,6 +198,14 @@ def initialize(user)
         user.persisted? || !script.login_required?
       end
     end
+    can :student_lesson_plan, Lesson do |lesson|
+      script = lesson.script
+      if script.pilot?
```

Comment:
I'm currently working out the tests for viewing teacher lesson plans when they are in a pilot script. Once I have that locked down we should probably add some tests checking that this is locked to only piloters when a script is in a pilot. https://github.com/code-dot-org/code-dot-org/pull/39208

Reply:
Just following up. I will own adding tests in my PR.